### DISCUSSION TITLE
Anvil

### DISCUSSION THREAD

**Discussion by opekope2** [opekope2/StaffMod#16] [Staff blocks]
Anvil is shown at [1:58-2:02](https://youtu.be/vlVUVsx6BYY?t=118) and [13:17-13:20](https://youtu.be/vlVUVsx6BYY?t=797), where the handlers couldn't lift it, because it's so heavy.

Attack damage: 40 (20 hearts), because that's the maximum amount of damage a [falling anvil](https://minecraft.wiki/w/Anvil#Falling_anvils) does + breaks shield (obviously), like an axe. 
Equip time: 4 seconds → DPS: 5 hearts
Movement speed: -100% (can be bypassed by jumping)

Other mechanic ideas:
* Just like falling anvils, anvils could break, when attacking.
* Chipped and damaged anvil could damage less
* Equip time could be reduced with strength status effect (which is easier said than done, but possible)
* Breaking mechanics: instant break up to a [certain hardness](https://minecraft.wiki/w/Breaking#Blocks_by_hardness) (to be discussed), but no break after that (because it's heavy, so you can't swing it)

**Comment by opekope2** [comment]
![CurseForge comment screenshot](https://github.com/opekope2/StaffMod/assets/20487556/fd646bee-9fb6-414d-a502-863c6e6060d1)

Another idea from CurseForge comments is to preduce or nullify jump strength. I don't yet know if I want to change vanilla mechanics this much.

**Reply by opekope2** [reply]
Preventing flight is very similar to it. Holding an anvil staff while flying would drag the player down

**Reply by opekope2** [reply]
Mojang just added some attribute modifiers in 24w06a, which make jump reduction possible without mixins. I'll definitely implement it once I port the mod to Minecraft 1.20.5.

**Comment by opekope2** [comment]
An idea I have is to instabreak blocks, which have less [hardness](https://minecraft.wiki/w/Breaking#Blocks_by_hardness) than a certain value (0.3, 0.4, or 0.5), and disable breaking blocks harder than that (because "Steve can't swing an anvil fast enough")

**Reply by charmander9695** [reply]
that is a smart idea

**Comment by thr3zims** [comment]
There seems to be a glitch with this. If you put it in your off-hand and back to your main hand, It no longer reduces your movement speed.

**Reply by opekope2** [reply]
I'm aware of this 😅, not sure why Minecraft does this. I'll look into it

**Reply by opekope2** [reply]
I was curious how long it takes to be noticed :shipit:

**Reply by FuryJack07** [reply]
How about instead of attributes, you use effects? Just check if a player is holding the anvil staff and then give them slowness 10

**Reply by opekope2** [reply]
If I use a different UUID for the offhand slowness, it solves the issue. No need for status effect. I'll fix it in the next release
